internal/pkg/workspace/workspace.go

```diff
@@ -318,6 +342,25 @@ func (ws *Workspace) CopilotDirPath() (string, error) {
 	}
 }
 
+// Workload is a description of what's associated with a given workload.
+type Workload struct {
+	WorkloadType string `yaml:"type"`
+}
+
+// isWorkloadType returns true if a manifest's type is one of the supported service types.
+func (ws *Workspace) isWorkloadType(mftBytes []byte, types []string) bool {
```

Comment:
With the recommendation above of taking a function as parameter to `workloadNames`, this can become just a type reader:
```go
func (ws *Workspace) readWorkloadType(dat []byte) (string, error)
```